webapp/channels/src/components/no_results_indicator/no_results_indicator.test.tsx

```diff
@@ -1,109 +1,133 @@
-            <NoResultsIndicator
-                iconGraphic={<div>{'Test'}</div>}
-                variant={NoResultsVariant.PinnedPosts}
-            />,
+    test('should render with variant PinnedPosts', () => {
+        render(
+            withIntl(
+                <NoResultsIndicator
+                    iconGraphic={<div>{'Test'}</div>}
+                    variant={NoResultsVariant.PinnedPosts}
+                    subtitleValues={{text: 'Pin to channel'}}
+                />,
+            ),
         );
 
-        expect(wrapper).toMatchSnapshot();
+        expect(screen.getByText('Test')).toBeInTheDocument();
+        expect(screen.getByText('No pinned messages yet')).toBeInTheDocument();
     });
 
-    test('should match snapshot with variant ChannelFiles', () => {
-        const wrapper = shallow(
-            <NoResultsIndicator
-                iconGraphic={<div>{'Test'}</div>}
-                variant={NoResultsVariant.ChannelFiles}
-            />,
+    test('should render with variant ChannelFiles', () => {
+        render(
+            withIntl(
+                <NoResultsIndicator
+                    iconGraphic={<div>{'Test'}</div>}
+                    variant={NoResultsVariant.ChannelFiles}
+                />,
+            ),
         );
 
-        expect(wrapper).toMatchSnapshot();
+        expect(screen.getByText('Test')).toBeInTheDocument();
+        expect(screen.getByText('No files yet')).toBeInTheDocument();
     });
 
-    test('should match snapshot with variant ChannelFilesFiltered', () => {
-        const wrapper = shallow(
-            <NoResultsIndicator
-                iconGraphic={<div>{'Test'}</div>}
-                variant={NoResultsVariant.ChannelFilesFiltered}
-            />,
+    test('should render with variant ChannelFilesFiltered', () => {
+        render(
+            withIntl(
+                <NoResultsIndicator
+                    iconGraphic={<div>{'Test'}</div>}
+                    variant={NoResultsVariant.ChannelFilesFiltered}
+                />,
+            ),
         );
 
-        expect(wrapper).toMatchSnapshot();
+        expect(screen.getByText('Test')).toBeInTheDocument();
+        expect(screen.getByText('No files found')).toBeInTheDocument();
     });
 
-    test('should match snapshot when expanded', () => {
-        const wrapper = shallow(
-            <NoResultsIndicator
-                variant={NoResultsVariant.Search}
-                expanded={true}
-            />,
+    test('should render when expanded', () => {
+        render(
+            withIntl(
+                <NoResultsIndicator
+                    variant={NoResultsVariant.Search}
+                    expanded={true}
+                    titleValues={{channelName: 'test-channel'}}
+                />,
+            ),
         );
 
-        expect(wrapper).toMatchSnapshot();
+        // Component renders in expanded state
+        expect(screen.getByText(/No results/)).toBeInTheDocument();
```

Comment:
Again, if we know the channel name (since we are passing it) we can be more specific about the assertion.